modules/components/src/utils/api.js

```diff
@@ -27,6 +28,7 @@ export const fetchExtendedMapping = ({ graphqlField, projectId }) =>
 
 export const addHeaders = headers => {
   alwaysSendHeaders = { ...alwaysSendHeaders, ...headers };
+  addDownloadHttpHeaders(headers);
```

Comment:
this is kind of messy in my opinion. Do we need to make these imperative calls? can't we just pass in headers when we actually make calls vs "setting" them and using the set ones later on?

Reply:
Added this issue so we can come back to this later :) : https://github.com/overture-stack/arranger/issues/323